usvg/src/image_href_resolver/mod.rs

```diff
@@ -0,0 +1,77 @@
+//! ImageHrefResolver specifies the way to parse `<image>` elements `xlink:href` value.
+
+pub mod default_resolver;
+
+use crate::ImageKind;
+use std::{
+    fmt::{Debug, Formatter, Result},
+    sync::Arc,
+};
+
+/// Functions that accept various representations of `xlink:href` value of the `<image`>
+/// element and return ImageKind that holds reference to the image buffer determined by these functions.
+#[derive(Clone, Debug)]
+pub struct ImageHrefResolver<'a> {
+    /// Resolver function that will be used if `xlink:href` is a DataUrl with encoded base64 string.
+    pub resolve_data: Box<dyn ImageHrefDataResolver + 'a>,
+    /// Resolver function that will be used to handle arbitrary string in `xlink:href`.
+    pub resolve_string: Box<dyn ImageHrefStringResolver + 'a>,
+}
+
+/// Image `href` resolver that accepts mime type string and reference to decoded base64 image data.
+pub trait ImageHrefDataResolver: Fn(&str, Arc<Vec<u8>>) -> Option<ImageKind> {
```

Comment:
Why do we need this monstrosity? Can't we simply have:

```rust
pub resolve_data: Box<dyn Fn(&str, Arc<Vec<u8>>) -> Option<ImageKind> + 'a>,
```

Reply:
All these shenanigans are to derive Debug and Clone traits, otherwise can't have closure as a part of `Options` struct.

Reply:
I would rather remove Debug and Clone from options.

Reply:
That works too 🙂

Reply:
Done.